app/javascript/components/tree-view/reducers/helpers.js

```diff
@@ -0,0 +1,21 @@
+import Tree from '@manageiq/react-ui-components/dist/wooden-tree';
+
+export const nodeCheckedWithDirty = (propNode, value) => {
+  let node = { ...propNode };
+  if (!node.state.disabled) {
+    if (!Object.hasOwnProperty.call(node.state, 'defaultChecked')) {
+      node = {
+        ...node,
+        state: {
+          ...node.state,
+          defaultChecked: node.state.checked,
+        },
+      };
+    }
+    node = {
+      ...Tree.nodeChecked(node, value),
```

Comment:
Is this a test for the node being checked? Or is this the function that actually checks the node? Just because using past tense for an operation that is happening right now can be confusing.

Reply:
That's a helper function from the tree. I won't change it as it introduces compatibility issues. It checks the node. In the tree, while writing it the naming gave a perfect sense.